src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -157,17 +157,15 @@ internal int EndRead(IAsyncResult asyncResult)
 
         internal IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback asyncCallback, object asyncState)
         {
-            return TaskToApm.Begin(WriteAsync(buffer, offset, count, CancellationToken.None), asyncCallback, asyncState);
+            return TaskToApm.Begin(WriteAsync(new ReadOnlyMemory<byte>(buffer, offset, count), CancellationToken.None), asyncCallback, asyncState);
```

Comment:
This can just be left as it was and pick up the validation done by the array-based WriteAsync.

Reply:
true but I had deleted that :) I guess if it needs to go back due to validation messages I will revert this as well.